saplings/profile/src/components/NewKey.js

```diff
@@ -36,53 +154,123 @@ export function NewKey() {
           </div>
           <div className="line" />
           <div className="key-form">
-            <form id="new-key-form" action="/profile">
-              <div className="input">
-                <label htmlFor="key-name">
-                  <div className="field-required">
-                    Key Name
-                    <div className="required-indicator">*</div>
-                  </div>
-                  <input type="text" id="key-name" name="key-name" required />
-                </label>
-              </div>
-              <div className="input">
-                <label htmlFor="public-key">
-                  <div className="field-required">
-                    Public Key
-                    <div className="required-indicator">*</div>
-                  </div> 
-                  <input type="text" id="public-key" name="public-key" required />
+            {!loading && (
+              <form id="new-key-form">
+                <div className="input">
+                  <label htmlFor="key-name">
+                    <div className="field-required">
+                      Key Name
+                      <div className="required-indicator">*</div>
+                    </div>
+                    <input
+                      type="text"
+                      id="key-name"
+                      name="name"
+                      value={state.name}
+                      onChange={handleChange}
+                      required
+                    />
                   </label>
+                </div>
+                <div className="input">
+                  <label htmlFor="public-key">
+                    <div className="field-required">
+                      Public Key
+                      <div className="required-indicator">*</div>
+                    </div>
+                    <input
+                      type="text"
+                      id="public-key"
+                      name="publicKey"
+                      value={state.publicKey}
+                      onChange={handleChange}
+                      required
+                      readOnly={readOnly}
+                    />
+                    </label>
+                </div>
+                <div className="input">
+                  <label htmlFor="private-key">
+                    <div className="field-required">
+                      Private Key
+                      <div className="required-indicator">*</div>
+                    </div>
+                    <div className="generate-key">
+                      <input
+                        type="password"
+                        id="private-key"
+                        name="privateKey"
+                        value={state.privateKey}
+                        onChange={handleChange}
+                        required
```

Comment:
Might make sense to also make the `private_key` field read-only if the keys are generated (as the `public_key` field is already)

Reply:
Agreed, there wouldn't be a reason to change just the private key after generating the pair, modified this input field to be read only if the key pair is generated